src/browser/core_state.js

```diff
@@ -499,6 +499,19 @@ function getSocketServerState() {
     return socketServerState;
 }
 
+/**
+ * Get app's very first ancestor
+ */
+function getAppAncestor(descendantAppUuid) {
+    const app = appByUuid(descendantAppUuid);
+
+    if (app && app.appObj && app.appObj.parentUuid) {
```

Comment:
This commit https://github.com/HadoukenIO/core/commit/a91bcbcd6957336594f95f9cfcda80ce4328a1ef